Bottom: 3e5d88814a08da355ad0f97933f3dde9575f6a9b
Top:    884c02ce5fcde834e46a5dc1316889b609f0280a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-21 16:25:08 +0100

Refresh of major-re-write-to-drawing-rout

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index bd9cf63..e877766 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1002,14 +1002,6 @@ ghid_screen_update (void)
 {
 }
 
-
-struct pin_info
-{
-  bool arg;
-  LayerTypePtr Layer;
-  const BoxType *drawn_area;
-};
-
 static int
 backE_callback (const BoxType * b, void *cl)
 {
@@ -1204,10 +1196,16 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon, const BoxType *dra
     }
 }
 
+struct poly_info
+{
+  LayerTypePtr Layer;
+  const BoxType *drawn_area;
+};
+
 static int
 poly_callback (const BoxType * b, void *cl)
 {
-  struct pin_info *i = (struct pin_info *) cl;
+  struct poly_info *i = (struct poly_info *) cl;
 
   DrawPlainPolygon (i->Layer, (PolygonTypePtr) b, i->drawn_area);
   return 1;
@@ -1217,14 +1215,10 @@ static int
 clearPin_callback (const BoxType * b, void *cl)
 {
   PinType *pin = (PinTypePtr) b;
-  struct pin_info *i = (struct pin_info *) cl;
-  if (i->arg)
-    {
-      if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
-        gui->thindraw_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
-      else
-        gui->fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
-    }
+  if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
+    gui->thindraw_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
+  else
+    gui->fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
   return 1;
 }
 
@@ -1241,9 +1235,7 @@ static int
 clearPin_callback_solid (const BoxType * b, void *cl)
 {
   PinTypePtr pin = (PinTypePtr) b;
-  struct pin_info *i = (struct pin_info *) cl;
-  if (i->arg)
-    gui->fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
+  gui->fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
   return 1;
 }
 
@@ -1259,28 +1251,24 @@ clearPad_callback_solid (const BoxType * b, void *cl)
 static void
 DrawMask (BoxType * screen)
 {
-  struct pin_info info;
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
 
   OutputType *out = &Output;
 
-  info.arg = true;
-  info.drawn_area = screen;
-
   if (thin)
     {
       gui->set_line_width (Output.pmGC, 0);
       gui->set_color (Output.pmGC, PCB->MaskColor);
-      r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback, &info);
-      r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback, &info);
-      r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback, &info);
+      r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback, NULL);
+      r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback, NULL);
+      r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback, NULL);
       gui->set_color (Output.pmGC, "erase");
     }
 
   gui->use_mask (HID_MASK_CLEAR);
-  r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback_solid, &info);
-  r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback_solid, &info);
-  r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback_solid, &info);
+  r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback_solid, NULL);
+  r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback_solid, NULL);
+  r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback_solid, NULL);
 
   gui->use_mask (HID_MASK_AFTER);
   gui->set_color (out->fgGC, PCB->MaskColor);
@@ -1296,7 +1284,7 @@ DrawLayerGroup (int group, const BoxType * screen)
 {
   int i, rv = 1;
   int layernum;
-  struct pin_info info;
+  struct poly_info info;
   LayerTypePtr Layer;
   int n_entries = PCB->LayerGroups.Number[group];
   Cardinal *layers = PCB->LayerGroups.Entries[group];
